api/src/services/taxonomy-service.ts

```diff
@@ -29,23 +32,25 @@ export class TaxonomyService {
    * @memberof TaxonomyService
    */
   async getTaxonByTsnIds(tsnIds: number[]): Promise<TaxonSearchResult[]> {
+    defaultLog.debug({ label: 'getTaxonByTsnIds', tsnIds });
+
     // Search for taxon records in the database
     const existingTaxonRecords = await this.taxonRepository.getTaxonByTsnIds(tsnIds);
-    let patchedTaxonRecords: TaxonRecord[] = [];
+    const existingTsnIds = existingTaxonRecords.map((record) => record.itis_tsn);
 
-    const missingTsnIds = tsnIds.filter((tsnId) => !existingTaxonRecords.find((item) => item.itis_tsn === tsnId));
+    const missingTsnIds = tsnIds.filter((tsnId) => !existingTsnIds.includes(tsnId));
 
     if (missingTsnIds.length) {
       // If the local database does not contain a record for all of the requested ids, search ITIS for the missing
       // taxon records, patching the missing records in the local database in the process
       const itisService = new ItisService();
       const itisResponse = await itisService.searchItisByTSN(missingTsnIds);
 
-      patchedTaxonRecords = await Promise.all(itisResponse.map(async (item) => this.addItisTaxonRecord(item)));
+      await Promise.all(itisResponse.map(this.addItisTaxonRecord));
     }
 
     // Missing ids patched, return taxon records for all requested ids
-    return this._sanitizeTaxonRecordsData(existingTaxonRecords.concat(patchedTaxonRecords));
+    return this._sanitizeTaxonRecordsData(await this.taxonRepository.getTaxonByTsnIds(tsnIds));
```

Comment:
Is the `addItisTaxonRecord` not working as expected? I thought the hope was that we could get the new record back after inserting (or conflicting and not inserting), so we wouldn't have to make another call afterwards (and can just combine the 2 results sets).

Reply:
From what I could gather, the query was not returning anything after running successfully 🤔

Reply:
Hmm weird, It seemed to be working for me in DBeaver, but maybe I need to try running it directly in PSQL just in case

Reply:
After some testing, addressing this comments appears to have worked successfully.